ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/ACSReader.scala

```diff
@@ -264,7 +264,9 @@ private[events] object FilterTableACSReader {
   ): () => ((TASK, Iterable[Long])) => Vector[Vector[Long]] = () => {
     val outputQueue = new BatchedDistinctOutputQueue(outputBatchSize)
     val taskQueue = new MergingTaskQueue[TASK](outputQueue.push)
-    val taskTracker = new TaskTracker[TASK](tasks, inputBatchSize)
+    // Limit the number of queued contract ids to 1M per filter
+    val maxTaskQueueSize = 1000000 / inputBatchSize
```

Comment:
How about making 1000000 configurable (with this default)? This can kill a stream.
What I am afraid about: if in production we figure that a huge stream can have glitches/rare big durations, which would quickly trigger the abort of the stream -- currently ACS is not restartable.

1000000 * (21 + 8) is still only 30megs, we could pour on this GBs of heap easily to let a huge ACS succeed (just in case)
(heap approximations from here: https://www.lihaoyi.com/post/BenchmarkingScalaCollections.html)

Reply:
I have added a CLI flag for configuring the queue limit, but I have left the default at 1M.